### PR TITLE
feat: use actor number and role in audit log

### PR DESCRIPTION
**Pull request by MadsDue**
## Description
Part 2 of peek/deque/requests audit log where we now use the two new fields in the audit log api to log the actor number and role.

From dev002 deployment:
image

## Checklist
- [ ] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [x] Has it been considered whether data is being delivered to the wrong actor?
- [x] Subsystem test executed (dev_002/dev_003) => https://github.com/Energinet-DataHub/dh3-environments/actions/runs/10829701865
- [x] Is there time to monitor state of the release to Production?
- [x] Reference to the task => https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/288

### CODE REVIEW

File: source/AuditLog/AuditLogClient/AuditLogHttpClient.cs [hunk 1]
```diff
@@ -41,6 +41,8 @@ public async Task LogAsync(
         Guid logId,
         Guid userId,
         Guid actorId,
+        string? actorNumber,
+        string? marketRoles,
```
Comment (RasmusGraabaek) [inline source/AuditLog/AuditLogClient/AuditLogHttpClient.cs:45]:
```suggestion
        string? marketRole,
```

I do believe that "market role" is a proper english word?

File: source/B2BApi.AppTests/OutgoingMessages/DequeueRequestListenerTests.cs [hunk 2]
```diff
@@ -129,7 +129,7 @@ public async Task Given_PersistedActorAndNoQeueue_When_CallingDequeueWithUnknown
         auditLogPayload.LogId.Should().NotBeEmpty();
         auditLogPayload.UserId.Should().Be(Guid.Empty);
         auditLogPayload.ActorId.Should().Be(Guid.Empty);
-        //auditLogPayload.ActorNumber.Should().Be(actorNumber);
+        auditLogPayload.ActorNumber.Should().Be(actorNumber.Value);
         auditLogPayload.SystemId.Should().Be(Guid.Parse("688b2dca-7231-490f-a731-d7869d33fe5e")); // EDI subsystem id
```
Comment (MWO1024) [inline source/B2BApi.AppTests/OutgoingMessages/DequeueRequestListenerTests.cs:133]: No check on roles?

File: source/B2BApi.AppTests/OutgoingMessages/PeekRequestListenerTests.cs [hunk 3]
```diff
@@ -138,7 +138,7 @@ public async Task Given_PersistedActorAndNoQeueue_When_CallingPeekAggregationsWi
         auditLogPayload.LogId.Should().NotBeEmpty();
         auditLogPayload.UserId.Should().Be(Guid.Empty);
         auditLogPayload.ActorId.Should().Be(Guid.Empty);
-        //auditLogPayload.ActorNumber.Should().Be(actorNumber);
+        auditLogPayload.ActorNumber.Should().Be(actorNumber.Value);
         auditLogPayload.SystemId.Should().Be(Guid.Parse("688b2dca-7231-490f-a731-d7869d33fe5e")); // EDI subsystem id
```
Comment (MWO1024) [inline source/B2BApi.AppTests/OutgoingMessages/PeekRequestListenerTests.cs:142]: No check on roles?

File: source/B2BApi.AppTests/IncomingMessages/IncomingMessageReceiverTests.cs [hunk 4]
```diff
@@ -144,7 +144,7 @@ public async Task Given_PersistedActor_When_CallingIncomingMessagesWithValidDocu
         auditLogPayload.LogId.Should().NotBeEmpty();
         auditLogPayload.UserId.Should().Be(Guid.Empty);
         auditLogPayload.ActorId.Should().Be(Guid.Empty);
-        //auditLogPayload.ActorNumber.Should().Be(actorNumber);
+        auditLogPayload.ActorNumber.Should().Be(actorNumber.Value);
         auditLogPayload.SystemId.Should().Be(Guid.Parse("688b2dca-7231-490f-a731-d7869d33fe5e")); // EDI subsystem id
```
[the remark this answers is not in the document]
Comment (MadsDue) [reply source/B2BApi.AppTests/IncomingMessages/IncomingMessageReceiverTests.cs:148]: Good catch

File: source/AuditLog/AuditLogClient/AuditLogHttpClient.cs (hunk 1 above)
Comment (MadsDue) [reply source/AuditLog/AuditLogClient/AuditLogHttpClient.cs:45]: it is a semicolon separated string :)

**Review by MWO1024 — APPROVED** [review]

**Pull request merged**